src/main/java/com/microsoft/lst_bench/common/DependentTaskExecutor.java

```diff
@@ -0,0 +1,132 @@
+ * first statement is stored in an intermediate object that can be specific to the connection. The
+ * expected object for a JDBC connection is of type List<Map<String, Object>>, handling of other
+ * objects would need to be added to the if-clause that checks the instance of the object.
+ */
+public class DependentTaskExecutor extends TaskExecutor {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(DependentTaskExecutor.class);
+  // TODO: Add functionality to use custom replacement markers.
+  private static final String DEFAULT_REPLACEMENT_MARKER = "dependent_replace";
+
+  private final Integer dependentBatchSize;
+
+  public DependentTaskExecutor(
+      SQLTelemetryRegistry telemetryRegistry,
+      String experimentStartTime,
+      Integer dependentBatchSize) {
+    super(telemetryRegistry, experimentStartTime);
+    // Set to a default of '1' if not otherwise specified.
+    if (dependentBatchSize == null) {
+      this.dependentBatchSize = 1;
+    } else {
+      this.dependentBatchSize = dependentBatchSize;
+    }
+  }
+
+  @Override
+  public void executeTask(Connection connection, TaskExec task, Map<String, Object> values)
+      throws ClientException {
+    for (FileExec file : task.getFiles()) {
+      Instant fileStartTime = Instant.now();
+      try {
+        for (int i = 0; i < file.getStatements().size(); i += 2) {
+          StatementExec statement = file.getStatements().get(i);
+
+          // Execute first query that retrieves the iterable input for the second query.
+          Instant statementStartTime = Instant.now();
+          Object iterableObject =
+              connection.executeQuery(
+                  StringUtils.replaceParameters(statement, values).getStatement());
+          writeStatementEvent(statementStartTime, statement.getId(), Status.SUCCESS);
+
+          if (iterableObject instanceof QueryResult) {
+            Map<String, List<Object>> valueList = ((QueryResult) iterableObject).getValueList();
+            statement = file.getStatements().get(i + 1);
+
+            // Determine the number of batchable values.
+            int size = 0;
+            for (Entry<String, List<Object>> pair : valueList.entrySet()) {
+              size = pair.getValue().size();
+              break;
+            }
+
+            // Execute second query repeatedly with the parameters extracted from the first query.
+            for (int j = 0; j < size; j += this.dependentBatchSize) {
+              int localMax =
+                  (j + this.dependentBatchSize) > size ? size : (j + this.dependentBatchSize);
+              Map<String, Object> localValues = new HashMap<>(values);
+              localValues.put(
+                  DEFAULT_REPLACEMENT_MARKER, this.createReplacementString(valueList, j, localMax));
+
+              statementStartTime = Instant.now();
+              connection.execute(
+                  StringUtils.replaceParameters(statement, localValues).getStatement());
```

Comment:
Do you need to pass all the values here? Wouldn't it be sufficient to pass the replacement that was created for the `DEFAULT_REPLACEMENT_MARKER`?

Reply:
The replacement is only the substring for the dependent part but there are also other parameters such as database or catalog that would need to be set, so I think this needs to be set somehow.